src/librbd/deep_copy/ObjectCopyRequest.cc

```diff
@@ -229,27 +227,32 @@ void ObjectCopyRequest<I>::handle_read_object(int r) {
 
 template <typename I>
 void ObjectCopyRequest<I>::send_read_from_parent() {
+  m_src_image_ctx->snap_lock.get_read();
+  m_src_image_ctx->parent_lock.get_read();
   io::Extents image_extents;
   compute_read_from_parent_ops(&image_extents);
+  m_src_image_ctx->snap_lock.put_read();
+
   if (image_extents.empty()) {
+    m_src_image_ctx->parent_lock.put_read();
     handle_read_from_parent(0);
     return;
   }
 
   ldout(m_cct, 20) << dendl;
-
-  assert(m_src_parent_image_ctx != nullptr);
+  assert(m_src_image_ctx->parent != nullptr);
 
   auto ctx = create_context_callback<
     ObjectCopyRequest<I>, &ObjectCopyRequest<I>::handle_read_from_parent>(this);
   auto comp = io::AioCompletion::create_and_start(
-    ctx, util::get_image_ctx(m_src_image_ctx), io::AIO_TYPE_READ);
+    ctx, util::get_image_ctx(m_src_image_ctx->parent), io::AIO_TYPE_READ);
   ldout(m_cct, 20) << "completion " << comp << ", extents " << image_extents
                    << dendl;
-  io::ImageRequest<I>::aio_read(m_src_parent_image_ctx, comp,
+  io::ImageRequest<I>::aio_read(m_src_image_ctx->parent, comp,
                                 std::move(image_extents),
                                 io::ReadResult{&m_read_from_parent_data}, 0,
                                 ZTracer::Trace());
+  m_src_image_ctx->parent_lock.put_read();
```

Comment:
@dillaman Using `m_src_image_ctx` at this point looks unsafe, as it may race with `handle_read_from_parent` calling `finish()`, in the case `aio_read` returns earlier due to a error. Reproducible with:
```
RBD_FEATURES=61 taskset --cpu-list 0 valgrind ./bin/unittest_librbd --gtest_filter=TestDeepCopy.CloneFlatten
```
Storing and using a local copy of `m_src_image_ctx` instead fixed it to me.

I believe this is the cause of the current jenkin's failure.